File: IE 211 Lab/src/Lab4/Sort.java
package Lab4;

public class Sort {
	
	final int bigM = 10000;
	
	public Sort() {}
	
	public void implementMergeSort(int[] array, int p, int r)
	{
		if(p < r)
		{
			int q = (p + r) / 2;
			
			//System.out.println(q);
			
			implementMergeSort(array, p, q);
			implementMergeSort(array, q+1, r);
			implementMerge(array, p, q, r);
		}
	}
	
	
	public void implementMerge(int[] array, int p, int q, int r)
	{
		int n1 = q-p+1;
		int n2 = r-q;
		
		int[] leftArray = new int[n1+1];
		int[] rightArray = new int[n2+1];
		
		for(int i = 0; i < n1; i++)
			leftArray[i] = array[p + i];
		
		leftArray[n1] = bigM;
		
		for(int j = 0; j < n2; j++)
			rightArray[j] = array[q + j + 1];
		
		rightArray[n2] = bigM;
		
//		System.out.println();
//		System.out.println("the left array is ");
//		
//		for(int i = 0; i < leftArray.length; i++)
//			System.out.print(leftArray[i] + " ");
//		
//		System.out.println();
//		System.out.println("the right array is ");
//		
//		for(int j = 0; j < rightArray.length; j++)
//			System.out.print(rightArray[j] + " ");
		
		int i = 0; int j = 0;
		
		for(int k = p; k < r + 1; k++)
			if(leftArray[i] <= rightArray[j])
			{
				array[k] = leftArray[i];
				i++;
			}
			else {
				array[k] = rightArray[j];
				j++;				
			}
		
	}
}

File: IE211 Fall22/src/Lab5/Sort.py
class Sort:

    runTime = 0
    bigM = 10000

    def __Merge(self, array, p, q, r):
        n1 = q - p + 1
        n2 = r - q

        leftArray = []
        rightArray = []

        for i in range(n1):
            leftArray.append(array[p + i])
        leftArray.append(self.bigM)
        print(f' The left array is {leftArray}')

        for j in range(n2):
            rightArray.append(array[q + j + 1])
        rightArray.append(self.bigM)
        print(f' The right array is {rightArray}')

        i = 0
        j = 0

        for k in range(p, r+1):
            if leftArray[i] <= rightArray[j]:
                array[k] = leftArray[i]
                i = i + 1

            else:
                array[k] = rightArray[j]
                j = j + 1

    def MergeSort(self, array, p, r):

        if(p < r):
            q = int((p + r) / 2)

            self.MergeSort(array, p , q)
            self.MergeSort(array, q + 1, r)
            self.__Merge(array, p, q, r)